### PR TITLE
add multi-file support

### PR DESCRIPTION
**Pull request by GusCaplan**

### CODE REVIEW

File: src/structures/interface/TextBasedChannel.js [hunk 1]
```diff
@@ -78,26 +78,34 @@ class TextBasedChannel {
...
         }
+        options.files[i] = file;
       }
 
-      return this.client.resolver.resolveBuffer(options.file.attachment).then(file =>
-        this.client.rest.methods.sendMessage(this, content, options, {
-          file,
-          name: options.file.name,
+      return Promise.all(options.files.map(file =>
+        this.client.resolver.resolveBuffer(file.attachment).then((buffer) => {
```
Comment (iCrawl) [inline src/structures/interface/TextBasedChannel.js]: remove parens around `buffer`

File: src/structures/interface/TextBasedChannel.js [hunk 2]
```diff
@@ -78,26 +78,34 @@ class TextBasedChannel {
...
+          file.file = buffer;
+          return file;
         })
-      );
+      )).then((files) => this.client.rest.methods.sendMessage(this, content, options, files));
```
Comment (iCrawl) [inline src/structures/interface/TextBasedChannel.js]: ditto around `files`

File: src/structures/interface/TextBasedChannel.js [hunk 3]
```diff
@@ -37,7 +37,7 @@ class TextBasedChannel {
    * (see [here](https://discordapp.com/developers/docs/resources/channel#embed-object) for more details)
    * @property {boolean} [disableEveryone=this.client.options.disableEveryone] Whether or not @everyone and @here
    * should be replaced with plain-text
-   * @property {FileOptions|string} [file] A file to send with the message
+   * @property {FileOptions[]|string[]} [files] Files to send with the message
```
Comment (hydrabolt) [inline src/structures/interface/TextBasedChannel.js]: I'd still keep the old file property because it can technically still be used

Comment (iCrawl) [inline src/structures/interface/TextBasedChannel.js]: done 👍

**Comment by GusCaplan** [comment]
i removed it because i want people to stop using it so we can phase it out in v12

**Comment by iCrawl** [comment]
Yes but this should just be marked as deprecated then.

**Pull request merged**